### PR TITLE
feat: Language Toggle cover all project

### PR DESCRIPTION
**Pull request by gustavop-dev**
# PR: Internationalization for Spanish and English

## Description

This pull request implements full project coverage for switching between Spanish and English. The key features include:

- **Language Toggle:** Added a language toggle switch that allows users to seamlessly switch between Spanish and English.
- **Translations:** Provided translations for all text elements across the project.
- **Reactivity:** Ensured that language changes are reactive and update the interface dynamically.

## Changes

- Integrated language switcher component in the main header.
- Updated all views and components to use translation keys.
- Provided translation files (`es.js` and `en.js`).
- Ensured compatibility with existing state management (Pinia) and other components.

### CODE REVIEW

File: frontend/src/components/ContactModel.vue [hunk 1]
```diff
@@ -9,7 +9,7 @@
                 <!-- Modal header -->
                 <div class="flex items-start justify-between p-5 pb-0">
                     <h3 class="text-2xl font-semibold text-black lg:text-4xl">
```
Comment (carlos18bp) [inline frontend/src/components/ContactModel.vue]: Add the `test-contact-getInTouch` class, please

File: frontend/src/components/layouts/Footer.vue [hunk 2]
```diff
@@ -10,8 +10,8 @@
                 <!-- Logo and company description -->
                 <div class="space-y-8">
                     <img src="@/assets/images/logo2.png" alt="Logo of Caropa Couture">
-                    <p class="text-md font-medium leading-6 text-white">
-                        Redefining fashion with exclusive,<br>fresh style and a unique personality.
+                    <p class="text-md font-medium leading-6 text-white test-text">
+                        {{ $t('text') }}
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: `text` very generic, you could use something more descriptive, like `footer-description` .
Don't forget to change the test class name if this variable changes

File: frontend/src/components/layouts/SearchBar.vue [hunk 3]
```diff
@@ -32,7 +32,7 @@
                 </div>
                 <!-- Link to see all products -->
                 <div class="mt-4 text-right">
-                    <RouterLink :to="{ name: 'catalog' }" @click="closeModal" class="font-medium me-6">See all products
+                    <RouterLink :to="{ name: 'catalog' }" @click="closeModal" class="font-medium me-6 test-searchBar-seeAllProducts">{{ $t('searchBar').seeAllProducts }}
```
Comment (carlos18bp) [inline frontend/src/components/layouts/SearchBar.vue]: Change to:
```
<RouterLink :to="{ name: 'catalog' }" @click="closeModal" class="font-medium me-6 test-searchBar-seeAllProducts">
     {{ $t('searchBar').seeAllProducts }}
</RouterLink>
```

File: frontend/src/components/product/AlertShoppingCart.vue [hunk 4]
```diff
@@ -1,10 +1,10 @@
 <template>
     <div v-if="visible" class="w-screen h-14 bg-primary flex justify-between items-center px-16 text-white font-regular text-md shadow-2xl">
         <div>
-            <p>Product added to shopping cart.</p>
+            <p class="test-addShoppingCart-text">{{ $t('addShoppingCart').text }}</p>
```
Comment (carlos18bp) [inline frontend/src/components/product/AlertShoppingCart.vue]: Change to:
```
<p class="test-addShoppingCart-text">
    {{ $t('addShoppingCart').text }}
</p>
```

File: frontend/src/components/product/AlertShoppingCart.vue [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
         </div>
         <div class="flex gap-6">
-            <a @click="shoppingCartToggle = true" class="cursor-pointer underline">See shopping cart</a>
+            <a @click="shoppingCartToggle = true" class="cursor-pointer underline test-addShoppingCart-seeCart">{{ $t('addShoppingCart').seeCart }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/product/AlertShoppingCart.vue]: Change to:
```
<a @click="shoppingCartToggle = true" class="cursor-pointer underline test-addShoppingCart-seeCart">
    {{ $t('addShoppingCart').seeCart }}
</a>
```

File: frontend/src/components/product/ShoppingCart.vue [hunk 6]
```diff
@@ -22,9 +22,9 @@
                     @removeProduct="removeProduct(product)" />
             </div>
             <div v-else class="text-lg font-regular ps-10">
-                <p>No products added</p>
+                <p>{{ $t('shoppingCart').noProducts }}</p>
                 <RouterLink :to="{ name: 'catalog' }" class="cursor-pointer">
-                    <span class="text-primary">Continue Shopping</span>
+                    <span class="text-primary test-shoppingCart-continueShopping">{{ $t('shoppingCart').continueShopping }}</span>
```
Comment (carlos18bp) [inline frontend/src/components/product/ShoppingCart.vue]: Change to:
```
<span class="text-primary test-shoppingCart-continueShopping">
    {{ $t('shoppingCart').continueShopping }}
</span>
```

File: frontend/src/views/AboutUs.vue [hunk 7]
```diff
@@ -5,39 +5,21 @@
...
             <!-- Text content -->
             <div>
-                <h2 class="text-2xl font-medium mb-4">WHO WE ARE</h2>
-                <p class="mb-4 font-regular">
-                    At Caropa Couture, we redefine confidence with every stitch. Our
-                    journey started with a vision to celebrate the unique beauty and
-                    strength of women; we strive to offer more than just stylish apparel.
-                    We believe in clothing that not only catches the eye but also lifts
-                    the spirit.
+                <h2 class="text-2xl font-medium mb-4 test-firstSection-whoWeAre-title">{{ $t('firstSection').whoWeAre.title }}</h2>
```
Comment (carlos18bp) [inline frontend/src/views/AboutUs.vue]: Change to:
```
<h2 class="text-2xl font-medium mb-4 test-firstSection-whoWeAre-title">
    {{ $t('firstSection').whoWeAre.title }}
</h2>
```

File: frontend/src/views/AboutUs.vue [hunk 8]
```diff
@@ -5,39 +5,21 @@
...
+                    {{ $t('firstSection').whoWeAre.paragraph }}
                 </p>
-                <h2 class="text-2xl font-medium mb-4">Our Story</h2>
-                <p class="mb-4 font-regular">
-                    Authenticity with Style <br>
-                    In the vibrant city of Miami, Florida, Caropa Couture is born under
-                    the vision of Carolin Martinez. This emerging brand aims to provide
-                    women with a fresh and uniquely styled option in their wardrobe,
-                    emphasizing confidence and authenticity in every piece.
+                <h2 class="text-2xl font-medium mb-4 test-firstSection-ourStory-title">{{ $t('firstSection').ourStory.title }}</h2>
```
Comment (carlos18bp) [inline frontend/src/views/AboutUs.vue]: Change to:
```
<h2 class="text-2xl font-medium mb-4 test-firstSection-ourStory-title">
     {{ $t('firstSection').ourStory.title }}
</h2>
```

File: frontend/src/views/Home.vue [hunk 9]
```diff
@@ -27,7 +27,7 @@
                             </span>
                             <span v-else>{{ category.title_es }}</span>
                         </h1>
-                        <h2 class="text-black font-light text-lg text-center">Shop Now</h2>
+                        <h2 class="text-black font-light text-lg text-center test-shopNow">{{ $t('shopNow') }}</h2>
```
Comment (carlos18bp) [inline frontend/src/views/Home.vue]: Change to:
```
<h2 class="text-black font-light text-lg text-center test-shopNow">
    {{ $t('shopNow') }}
</h2>
```

File: frontend/src/views/Home.vue [hunk 10]
```diff
@@ -108,7 +107,7 @@
                             </span>
                             <span v-else>{{ category.title_es }}</span>
                         </h1>
-                        <h2 class="text-black font-light text-md text-center">Shop Now</h2>
+                        <h2 class="text-black font-light text-md text-center test-shopNow">{{ $t('shopNow') }}</h2>
```
Comment (carlos18bp) [inline frontend/src/views/Home.vue]: Change to:
```
<h2 class="text-black font-light text-md text-center test-shopNow">
   {{ $t('shopNow') }}
</h2>
```

File: frontend/src/views/product/Checkout.vue [hunk 11]
```diff
@@ -15,11 +15,11 @@
             }}</span>
                 </div>
                 <div class="flex items-center space-x-2 text-lg">
-                    <span class="cursor-pointer font-bold" @click="handleLanguage('en')">
+                    <span :class="{ 'border-b-2 border-black border-current': currentLanguage === 'en' }" class="cursor-pointer font-bold" @click="handleLanguage('en')">
                         EN
                     </span>
```
Comment (carlos18bp) [inline frontend/src/views/product/Checkout.vue]: Change to:
```
<span :class="{ 'border-b-2 border-black border-current': currentLanguage === 'en' }" 
           class="cursor-pointer font-bold" 
           @click="handleLanguage('en')">
   EN
</span>
```

File: frontend/src/views/product/Checkout.vue [hunk 12]
```diff
@@ -15,11 +15,11 @@
...
                     <span class="font-bold">|</span>
-                    <span class="cursor-pointer font-bold" @click="handleLanguage('es')"> ES </span>
+                    <span :class="{ 'border-b-2 border-black border-current': currentLanguage === 'es' }" class="cursor-pointer font-bold" @click="handleLanguage('es')"> ES </span>
```
Comment (carlos18bp) [inline frontend/src/views/product/Checkout.vue]: Change to:
```
<span :class="{ 'border-b-2 border-black border-current': currentLanguage === 'es' }" 
           class="cursor-pointer font-bold" 
           @click="handleLanguage('es')"> 
    ES
</span>
```

**Review by carlos18bp — CHANGES_REQUESTED** [review]

File: frontend/src/views/product/Checkout.vue (hunk 12 above)
Comment (gustavop-dev) [reply frontend/src/views/product/Checkout.vue]: Done!

File: frontend/src/views/product/Checkout.vue (hunk 11 above)
Comment (gustavop-dev) [reply frontend/src/views/product/Checkout.vue]: Done!

File: frontend/src/views/Home.vue (hunk 10 above)
Comment (gustavop-dev) [reply frontend/src/views/Home.vue]: Done!

File: frontend/src/views/Home.vue (hunk 9 above)
Comment (gustavop-dev) [reply frontend/src/views/Home.vue]: Done!

File: frontend/src/views/AboutUs.vue (hunk 8 above)
Comment (gustavop-dev) [reply frontend/src/views/AboutUs.vue]: Done!

File: frontend/src/views/AboutUs.vue (hunk 7 above)
Comment (gustavop-dev) [reply frontend/src/views/AboutUs.vue]: Done!

File: frontend/src/components/product/ShoppingCart.vue (hunk 6 above)
Comment (gustavop-dev) [reply frontend/src/components/product/ShoppingCart.vue]: Done!

File: frontend/src/components/product/AlertShoppingCart.vue (hunk 5 above)
Comment (gustavop-dev) [reply frontend/src/components/product/AlertShoppingCart.vue]: Done!

File: frontend/src/components/product/AlertShoppingCart.vue (hunk 4 above)
Comment (gustavop-dev) [reply frontend/src/components/product/AlertShoppingCart.vue]: Done!

File: frontend/src/components/layouts/SearchBar.vue (hunk 3 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/SearchBar.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 2 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/ContactModel.vue (hunk 1 above)
Comment (gustavop-dev) [reply frontend/src/components/ContactModel.vue]: Done!

File: frontend/src/components/ProductCarousel.vue [hunk 13]
```diff
@@ -2,10 +2,10 @@
     <!-- Carousel container for trending products -->
     <div v-if="productsOnTrending" class="carousel-container px-8 max-w-7xl mx-auto py-16">
         <div class="text-center mb-8">
-            <h2 class="text-3xl font-semibold">TRENDING NOW</h2>
+            <h2 class="text-3xl font-semibold test-thirdSection-title">{{ $t('thirdSection').title }}</h2>
```
Comment (carlos18bp) [inline frontend/src/components/ProductCarousel.vue]: Change to:
```
<h2 class="text-3xl font-semibold test-thirdSection-title">
    {{ $t('thirdSection').title }}
</h2> 
```

File: frontend/src/components/layouts/Footer.vue [hunk 14]
```diff
@@ -34,20 +34,20 @@
                 <div class="mt-16 grid grid-cols-2 gap-8 xl:col-span-2 xl:mt-0">
                     <div class="md:grid md:grid-cols-2 md:gap-8">
                         <div>
-                            <h3 class="text-sm font-medium leading-6 text-white">COMPANY</h3>
+                            <h3 class="text-sm font-medium leading-6 text-white test-company">{{ $t('company') }}</h3>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: Change to:
```
<h3 class="text-sm font-medium leading-6 text-white test-company">
   {{ $t('company') }}
</h3>
```

File: frontend/src/components/layouts/Footer.vue [hunk 15]
```diff
@@ -34,20 +34,20 @@
...
                             <ul role="list" class="mt-6 grid grid-cols-1 gap-4 font-regular text-white text-sm">
                                 <li>
-                                    <a @click="goTo('catalog')" class="cursor-pointer">Shop</a>
+                                    <a @click="goTo('catalog')" class="cursor-pointer test-shop">{{ $t('shop') }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: Change to:
```
<a @click="goTo('catalog')" class="cursor-pointer test-shop">
   {{ $t('shop') }}
</a>
```

File: frontend/src/components/layouts/Footer.vue [hunk 16]
```diff
@@ -34,20 +34,20 @@
...
                                 </li>
                                 <li>
-                                    <a @click="goTo('about_us')" class="cursor-pointer">About Us</a>
+                                    <a @click="goTo('about_us')" class="cursor-pointer test-about">{{ $t('about') }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: Change to: 
```
<a @click="goTo('about_us')" class="cursor-pointer test-about">
   {{ $t('about') }}
</a>
```

File: frontend/src/components/layouts/Footer.vue [hunk 17]
```diff
@@ -56,7 +56,7 @@
                             </ul>
                         </div>
                         <div class="mt-10 md:mt-0">
-                            <h3 class="text-sm font-medium leading-6 text-white">NAVIGATE</h3>
+                            <h3 class="text-sm font-medium leading-6 text-white test-navigate">{{ $t('navigate') }}</h3>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: Change to:
```
<h3 class="text-sm font-medium leading-6 text-white test-navigate">
   {{ $t('navigate') }}
</h3>
```

File: frontend/src/components/layouts/Footer.vue [hunk 18]
```diff
@@ -71,7 +71,7 @@
                             <h3 class="text-sm font-semibold leading-6 text-white">LEGAL</h3>
                             <ul role="list" class="mt-6 space-y-4 font-regular text-white text-sm">
                                 <li>
-                                    <a href="#" class="leading-6">Terms & Conditions</a>
+                                    <a href="#" class="leading-6 test-termsConditions">{{ $t('termsConditions') }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: Same above

File: frontend/src/components/layouts/Footer.vue [hunk 19]
```diff
@@ -81,7 +81,7 @@
             
             <!-- Footer bottom section -->
             <div class="mt-16 border-t border-white/10 pt-8 sm:mt-20 lg:mt-24">
-                <p class="text-sm font-medium leading-5 text-white">&copy; 2020 Caropa Couture. All rights reserved</p>
+                <p class="text-sm font-medium leading-5 text-white test-rights">{{ $t('rights') }}</p>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Footer.vue]: Same above

File: frontend/src/components/layouts/Header.vue [hunk 20]
```diff
@@ -7,11 +7,11 @@
...
                 <!-- Navigation links -->
-                <a @click="goTo('catalog')" class="cursor-pointer">Shop</a>
-                <a @click="showContactModel = true" class="cursor-pointer">
-                    Contact
+                <a @click="goTo('catalog')" class="cursor-pointer test-header-shop">{{ $t('header').shop }}</a>
+                <a @click="showContactModel = true" class="cursor-pointer test-header-contact">
+                    {{ $t('header').contact }}
                 </a>
-                <a @click="goTo('about_us')" class="cursor-pointer">About</a>
+                <a @click="goTo('about_us')" class="cursor-pointer test-header-about">{{ $t('header').about }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Header.vue:14]: Same above

File: frontend/src/components/layouts/Header.vue [hunk 21]
```diff
@@ -32,7 +32,7 @@
                 <div>
                     <div @click="showSearchBar = true" class="flex items-center justify-center gap-3 cursor-pointer">
                         <MagnifyingGlassIcon class="text-black size-6"></MagnifyingGlassIcon>
-                        <a class="text-md font-famil-semibold text-black">Search</a>
+                        <a class="text-md font-famil-semibold text-black test-header-search">{{ $t('header').search }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Header.vue]: Same above

File: frontend/src/components/layouts/Header.vue [hunk 22]
```diff
@@ -111,11 +111,11 @@
                         {{ category }}
                     </a>
                     <!-- Navigation links -->
-                    <a @click="goTo('catalog')" class="pt-2 cursor-pointer border-t border-t-gray-500 font-regular">Shop</a>
-                    <a @click="showContactModel = true" class="cursor-pointer font-regular">
-                        Contact
+                    <a @click="goTo('catalog')" class="pt-2 cursor-pointer border-t border-t-gray-500 font-regular test-header-shop">{{ $t('header').shop }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Header.vue]: Same above

File: frontend/src/components/layouts/Header.vue [hunk 23]
```diff
@@ -111,11 +111,11 @@
...
+                    <a @click="showContactModel = true" class="cursor-pointer font-regular test-header-contact">
+                        {{ $t('header').contact }}
                     </a>
-                    <a @click="goTo('about_us')" class="cursor-pointer font-regular">About</a>
+                    <a @click="goTo('about_us')" class="cursor-pointer font-regular test-header-about">{{ $t('header').about }}</a>
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Header.vue]: Same above

File: frontend/src/components/layouts/SearchBar.vue [hunk 24]
```diff
@@ -9,7 +9,7 @@
             <div class="flex items-center justify-between border-b pb-4">
                 <input type="text"
                     class="w-full p-2 text-lg border-transparent focus:border-none focus:ring-0 focus:outline-none font-semibold"
-                    placeholder="Search..." v-model="searchQuery" @input="onSearch(searchQuery)" />
+                    :placeholder="$t('searchBar').search" v-model="searchQuery" @input="onSearch(searchQuery)" />
```
Comment (carlos18bp) [inline frontend/src/components/layouts/SearchBar.vue]: Change to:
```
<input type="text"
class="w-full p-2 text-lg border-transparent focus:border-none focus:ring-0 focus:outline-none font-semibold"
 :placeholder="$t('searchBar').search"
v-model="searchQuery" 
@input="onSearch(searchQuery)" />
```

File: frontend/src/components/product/ShoppingCart.vue [hunk 25]
```diff
@@ -8,7 +8,7 @@
         <div ref="cart" class="relative bg-white h-full w-full shadow-lg flex flex-col z-60 lg:w-2/5">
             <!-- Cart Header -->
             <div class="flex justify-between items-center p-10">
-                <h2 class="text-2xl font-famil-semibold">Shopping Cart</h2>
+                <h2 class="text-2xl font-famil-semibold">{{ $t('shoppingCart').shoppingCart }}</h2>
```
Comment (carlos18bp) [inline frontend/src/components/product/ShoppingCart.vue]: Same above

File: frontend/src/views/product/Detail.vue [hunk 26]
```diff
@@ -70,7 +70,7 @@
 
                         <!-- Size Options -->
                         <div class="mb-12">
-                            <h3 class="text-md font-medium text-gray-700">Size</h3>
+                            <h3 class="text-md font-medium text-gray-700 test-size">{{ $t('size') }}</h3>
```
Comment (carlos18bp) [inline frontend/src/views/product/Detail.vue]: Same above

**Review by carlos18bp — CHANGES_REQUESTED** [review]

File: frontend/src/components/ProductCarousel.vue (hunk 13 above)
Comment (gustavop-dev) [reply frontend/src/components/ProductCarousel.vue]: Done!

Comment (gustavop-dev) [reply frontend/src/components/ProductCarousel.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 14 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 15 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 16 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 17 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 18 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/layouts/Footer.vue (hunk 19 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Footer.vue]: Done!

File: frontend/src/components/layouts/Header.vue (hunk 20 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Header.vue:14]: Done!

File: frontend/src/components/layouts/Header.vue (hunk 21 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Header.vue]: Done!

File: frontend/src/components/layouts/Header.vue (hunk 22 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Header.vue]: Done!

File: frontend/src/components/layouts/Header.vue (hunk 23 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Header.vue]: Done!

File: frontend/src/components/layouts/SearchBar.vue (hunk 24 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/SearchBar.vue]: Done!

File: frontend/src/components/product/ShoppingCart.vue (hunk 25 above)
Comment (gustavop-dev) [reply frontend/src/components/product/ShoppingCart.vue]: Done!

File: frontend/src/views/product/Detail.vue (hunk 26 above)
Comment (gustavop-dev) [reply frontend/src/views/product/Detail.vue]: Done!

**Review by carlos18bp — APPROVED** [review]

**Pull request merged**